### PR TITLE
Collapsing action tool bar with android material design in Kotlin

### LINKED ISSUE #5333: [Android] Collapsing Action-Tool-Bar with Android Material Design  in  Kotlin
### Topic Suggestion
Collapsing Action-Tool-Bar with Android Material Design  in  Kotlin

#### Proposed article introduction
In Material Design API level 30, the Android Toolbar is added as an Action Bar in the intended Activity. The activity title, navigation icons, and other views can be displayed in the Android toolbar.  Collapsing toolbar is a more advanced variant of the conventional toolbar widget that is used in conjunction with the Collapsing Toolbar Layout widget. This widget allows us to build an incredible toolbar motion called expanding and collapsing toolbar on-screen scrolling. This feature will raise the toolbar size that was initially specified in the App Bar Layout widget automatically. 

In this tutorial, we will discuss more on collapsing action toolbar with the android material design using a  step-by-step instruction for the Android Material Design Collapsing using a nested scrollable layout.

#### Key takeaways 
- Overview on what is Collapsing Action-Tool-Bar
- Step by Step Implementation
  1.  Create a New Project
  2. Adding application dependencies
  3. Adding a menu toolbar
  4.  Creating activity_main.xml layout file.
-  Testing our application 

#### Article quality
This will be a unique tutorial that will include a step-by-step explanation to allow all the readers to have a better understanding of the concepts used. It will also include an illustration (Application) to allow the reader to grasp the content covered easily.

#### References
N/A

### PR DESCRIPTION
**Pull request by dvoredave**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5333 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/authors/author/index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (Jethro-magaji) [inline content/authors/author/index.md:1]: Please edit your author bio in the proper Markdown format. Here's an [example](https://raw.githubusercontent.com/section-engineering-education/engineering-education/master/content/authors/adith-bharadwaj/index.md)

File: content/articles/collapsing-action-tool-bar-with-android-material-design-in-kotlin/index.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+This tutorial will build an application that utilizes the Collapsing Toolbar layout widget. We shall also include Android's new design support library to create scrolling effects that are eye-catching animations(palette API). There will be precise step-by-step procedures that will be easy to follow and understand for beginners and intermediate.
+
+### Table of contents
+- [Setting up the project environment](#setting-up-the-project-environment)
+- [Adding the appropriate dependencies](#adding-the-appropriate-dependencies)
+- [Designing the XML layout](#designing-the-xml-layout)
+- [Writing Kotlin code](#writing-kotlin-code)
+- [Testing the application](#testing-the-application)
+- [Conclusion](#conclusion)
+
```
Comment (Jethro-magaji) [inline content/articles/collapsing-action-tool-bar-with-android-material-design-in-kotlin/index.md:13]: Add a Prerequisites' section here, to provide the audience with any prerequisites that may be required to follow your tutorial. For example: Need basic HTML and CSS knowledge for an article on how to use the CSS pre-processor, Sass.

File: content/articles/collapsing-action-tool-bar-with-android-material-design-in-kotlin/index.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    <gradient
+        android:angle="90"
+        android:endColor="@android:color/transparent"
+        android:startColor="@color/colorDarkSemiTransparent"/>
+</shape>
+```
+
+The actual layout of the title background should look like the one shown below.
+
+![Title background](engineering-education/collapsing-action-tool-bar-with-android-material-design-in-kotlin/title-bg.png)
```
Comment (Jethro-magaji) [inline content/articles/collapsing-action-tool-bar-with-android-material-design-in-kotlin/index.md:80]: Like this `/engineering-education/` not this `engineering-education/`

File: content/articles/collapsing-action-tool-bar-with-android-material-design-in-kotlin/index.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+                collapsingToolbar.setContentScrimColor(palette.getVibrantColor(R.attr.colorPrimary))
+            }
+        }
+```
+
+To use the Palette API, you must first supply it with a bitmap. The API then uses the picture to produce colors, i.e., `Palette.from(bitmap).generate ...`.
+Then we use the if statement to find whether the image has a primary color. If it has the main color(primary color), the color is set to be the content scrim color of the toolbar.
+
+### Testing the application
+We need an emulator(virtual device) installed in the computer or an actual physical device such as a mobile phone or a phone tablet.
```
Comment (Jethro-magaji) [inline content/articles/collapsing-action-tool-bar-with-android-material-design-in-kotlin/index.md:246]: `Emulator (virtual device)` should be listed in the prerequisites section.

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi! @auxesis 
Well done for such an amazing article. However, there are some corrections you need to make.
Please, run your article on tools like Grammarly and ProWritingAid for better editing. This would save the review time to review your work.
Here's a [link](https://github.com/section-engineering-education/engineering-education/pull/6210/commits/acc44893d4ec258f4e077b9ce8e3e2ca2b9d6a9f) to a PR with little or no revision to guide you on formatting. If you have any questions or need help, you can reach out to me on the EngEd Slack channel.
Thanks.

**Comment by dvoredave** [comment]
Hello @Jethro-magaji, I made the changes as requested. Thank you.

**Comment by hectorkambow** [comment]
@dvoredave 
have you gone through the entire article to check for other mistakes? Our peer reviewers are not expected to POINT OUT everything. 👍

**Comment by Jethro-magaji** [comment]
Hello @dvoredave 
Seems like you have 2 accounts on Section EngEd, and it's causing a merge conflict.
Please upload a high resolution free stock hero image 600x400px.

**Review by Jethro-magaji — APPROVED** [review]
Hi! @dvoredave 
Well done for coming up with such an amazing article. Please take note of these corrections to minimize revision next time.
Thanks.

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@dvoredave 

Can you please DM via Slack asap

**Comment by hectorkambow** [comment]
@davidginmaina

**Comment by hectorkambow** [comment]
Why do you both have the same name again?

**Pull request merged**